File: app/src/main/java/com/example/sylhetjerseyhouse/ItemView.java
package com.example.sylhetjerseyhouse;

import androidx.appcompat.app.AppCompatActivity;
import androidx.room.Room;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;
import com.example.sylhetjerseyhouse.db.AppDatabase;
import com.example.sylhetjerseyhouse.db.Item;
import com.example.sylhetjerseyhouse.db.ItemDAO;

public class ItemView extends AppCompatActivity {

    TextView title;
    TextView description;
    TextView price;
    TextView quantity;
    ImageView image_view;
    ImageButton increment, decrement;
    Button addToCart;
    Integer s_quantity, JerseyPrice;
    String s_title, s_price, s_description, s_image;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_item_view);




        title = findViewById(R.id.id_title);
        price = findViewById(R.id.id_price);
        description = findViewById(R.id.id_description);
        image_view = findViewById(R.id.id_image);
        quantity = findViewById(R.id.id_quantity);
        increment = findViewById(R.id.btn_plus);
        decrement = findViewById(R.id.btn_minus);
        addToCart = findViewById(R.id.id_addToCartbtn);

        Intent data = getIntent();
        s_title = data.getStringExtra("title");
        s_price = data.getStringExtra("price");
        s_description = data.getStringExtra("description");
        s_image = data.getStringExtra("imageData");

        try {
            JerseyPrice = Integer.parseInt(s_price);
        } catch (NumberFormatException e) {
            System.err.println(e.getMessage());
        }


        title.setText(s_title);
        price.setText(s_price);
        description.setText(s_description);
        Glide.with(this)
                .load(s_image)
                .centerCrop()
                .placeholder(R.mipmap.ic_launcher)
                .into(image_view);


        increment.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String currentValue = quantity.getText().toString();
                int value = Integer.parseInt(currentValue);
                value++;
                int yourPrice = value * JerseyPrice;
                price.setText(String.valueOf(yourPrice));
                quantity.setText(String.valueOf(value));
            }
        });

        decrement.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String currentValue = quantity.getText().toString();
                int value = Integer.parseInt(currentValue);
                if (value > 1) {
                    value--;
                    int yourPrice = value * JerseyPrice;
                    price.setText(String.valueOf(yourPrice));
                    quantity.setText(String.valueOf(value));
                }
            }
        });


        addToCart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                try {

                    s_quantity = Integer.parseInt(quantity.getText().toString());
                    int id = (int) Math.random();

                    AppDatabase db = Room.databaseBuilder(getApplicationContext(), AppDatabase.class, "cart_db").allowMainThreadQueries().build();
                    ItemDAO itemDAO = db.itemDAO();
                    Boolean check = itemDAO.is_exist(s_title);
                    if (check == false) {
                        try {
                            itemDAO.insertRecord(new Item(id, s_title, s_price, s_quantity));
                            Toast.makeText(ItemView.this, "Item added to cart", Toast.LENGTH_SHORT).show();
                        } catch (Exception e) {
                            Toast.makeText(ItemView.this, "" + e, Toast.LENGTH_SHORT).show();
                        }
                    } else {
                        Toast.makeText(ItemView.this, "Item Already exist in cart", Toast.LENGTH_SHORT).show();
                    }

                }catch (Exception e)
                {
                    Toast.makeText(ItemView.this, ""+e, Toast.LENGTH_SHORT).show();
                }


            }
        });




    }
}

File: app/src/main/java/com/eshopthis/finds/Activities/ItemView.kt
package com.eshopthis.finds.Activities

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.room.Room
import com.bumptech.glide.Glide
import com.eshopthis.finds.R
import com.eshopthis.finds.data.AppDatabase
import com.eshopthis.finds.data.Item
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class ItemView : AppCompatActivity() {

    private lateinit var title: TextView
    private lateinit var description: TextView
    private lateinit var price: TextView
    private lateinit var quantity: TextView
    private lateinit var image_view: ImageView
    private lateinit var increment: ImageButton
    private lateinit var decrement: ImageButton
    private lateinit var addToCart: Button
    private var s_quantity: Int = 1
    private var JerseyPrice: Double = 0.0
    private var s_title: String = ""
    private var s_price: String = ""
    private var s_description: String = ""
    private var s_image: String = ""
    private var id: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_item_view)

        title = findViewById(R.id.id_title)
        price = findViewById(R.id.id_price)
        description = findViewById(R.id.id_description)
        image_view = findViewById(R.id.id_image)
        quantity = findViewById(R.id.id_quantity)
        increment = findViewById(R.id.btn_plus)
        decrement = findViewById(R.id.btn_minus)
        addToCart = findViewById(R.id.id_addToCartbtn)

        val data: Intent? = intent
        s_title = data?.getStringExtra("title") ?: ""
        s_price = data?.getStringExtra("price") ?: ""
        s_description = data?.getStringExtra("description") ?: ""
        s_image = data?.getStringExtra("imageData") ?: ""

        try {
            JerseyPrice = s_price.toDouble()
        } catch (e: NumberFormatException) {
            System.err.println(e.message)
        }

        title.text = s_title
        price.text = s_price
        description.text = s_description
        Glide.with(this)
            .load(s_image)
            .centerCrop()
            .placeholder(R.mipmap.ic_launcher)
            .into(image_view)

        increment.setOnClickListener {
            s_quantity++
            val yourPrice = s_quantity * JerseyPrice
            price.text = yourPrice.toString()
            quantity.text = s_quantity.toString()
        }

        decrement.setOnClickListener {
            if (s_quantity > 1) {
                s_quantity--
                val yourPrice = s_quantity * JerseyPrice
                price.text = yourPrice.toString()
                quantity.text = s_quantity.toString()
            }
        }

        addToCart.setOnClickListener {
            uploadImage()
        }
    }

    private fun uploadImage() {
        try {
            id = (Math.random() * Integer.MAX_VALUE).toInt()

            GlobalScope.launch(Dispatchers.IO) {
                val db = Room.databaseBuilder(applicationContext, AppDatabase::class.java, "cart_db").allowMainThreadQueries().build()
                val itemDAO = db.itemDao()
                val check = itemDAO.isTitleExists(s_title)
                withContext(Dispatchers.Main) {
                    if (check == 0) {
                        itemDAO.insertItem(Item(id = id, name = s_title, description = s_description, price = JerseyPrice, quantity = s_quantity))
                        Toast.makeText(this@ItemView, "Item added to cart", Toast.LENGTH_SHORT).show()
                    } else {
                        Toast.makeText(this@ItemView, "Item Already exist in cart", Toast.LENGTH_SHORT).show()
                    }
                }
            }
        } catch (e: Exception) {
            Toast.makeText(this@ItemView, e.toString(), Toast.LENGTH_SHORT).show()
        }
    }
}